src/System.Linq.Parallel/src/System/Linq/ParallelEnumerable.cs

```diff
@@ -4390,20 +4390,27 @@ public static bool SequenceEqual<TSource>(this ParallelQuery<TSource> first, Par
             QuerySettings settings = leftOp.SpecifiedQuerySettings.Merge(rightOp.SpecifiedQuerySettings)
                 .WithDefaults()
                 .WithPerExecutionSettings(new CancellationTokenSource(), new System.Linq.Parallel.Shared<bool>(false));
+            
+            // If an operator is from "data" - ie, Enumerable.AsParallel() - it will usually wrap exceptions (w/ AggregateException)
+            // by the indirection present in QueryExecutionOption.GetEnumerator().
+            bool firstBare = leftOp is ScanQueryOperator<TSource>;
+            bool secondBare = rightOp is ScanQueryOperator<TSource>;
 
             // If first.GetEnumerator throws an exception, we don't want to wrap it with an AggregateException.
-            IEnumerator<TSource> e1 = first.GetEnumerator();
+            IEnumerator<TSource> e1 = new QueryExecutionOption<TSource>(leftOp, rightOp.SpecifiedQuerySettings).GetEnumerator();
             try
             {
+                bool firstThrew = false;
+                bool secondThrew = false;
                 // If second.GetEnumerator throws an exception, we don't want to wrap it with an AggregateException.
-                IEnumerator<TSource> e2 = second.GetEnumerator();
+                IEnumerator<TSource> e2 = new QueryExecutionOption<TSource>(rightOp, leftOp.SpecifiedQuerySettings).GetEnumerator();
```

Comment:
This wrapping with QueryExecutionOption is now adding allocations and an additional operation in the chain through which all calls will need to proceed.  That could add measurable overhead.